osquery/tables/networking/posix/interfaces.cpp

```diff
@@ -187,7 +229,25 @@ void genDetailsFromAddr(const struct ifaddrs* addr, QueryData& results) {
     r["odrops"] = INTEGER(0);
     r["collisions"] = BIGINT_FROM_UINT32(ifd->ifi_collisions);
     r["last_change"] = BIGINT_FROM_UINT32(ifd->ifi_lastchange.tv_sec);
-#endif
+    if (context.isColumnUsed("link_speed")) {
+      int fd = socket(AF_INET, SOCK_DGRAM, 0);
+      if (fd >= 0) {
+        struct ifmediareq ifmr = {};
+        memcpy(ifmr.ifm_name, addr->ifa_name, sizeof(ifmr.ifm_name));
+        ifmr.ifm_ulist = new int[ifmr.ifm_count];
```

Comment:
This is still heap allocated but won't auto free now. If ifm_count is going to be small then we should just allocate it on the stack, but otherwise the uniqueptr should work and is a fine solution.

You can leaks check by running:  ./tools/analysis/profile.py --leaks --query "..."